packages/adapter/addon/-private/utils/determine-body-promise.ts

```diff
@@ -3,8 +3,10 @@ import { DEBUG } from '@glimmer/env';
 
 import continueOnReject from './continue-on-reject';
 
+type Dict<T> = import('@ember-data/store/-private/ts-interfaces/utils').Dict<T>;
+
 type RequestData = import('../../rest').RequestData;
-type Payload = object | string | undefined;
+type Payload = Dict<unknown> | string | undefined;
```

Comment:
Is it guaranteed to not be an array?

Reply:
unsure, but `object` didn't cover anything but literally `Object` itself, can check if this should also include array.

Reply:
since JSON.parse can parse a top level array then yes, so I've added `unknown[]` and made fixes to types accordingly.